### PR TITLE
OP-777 admission and discharge time

### PR DESCRIPTION
**Pull request by mwithi**
See OP-777.

Also fixed operation date format and another report (static parameters) and

### CODE REVIEW

File: src/main/java/org/isf/opd/gui/OpdBrowser.java [hunk 1]
```diff
@@ -381,7 +381,7 @@ private JButton getJDeleteButton() {
 				Opd opd = (Opd) (model.getValueAt(jTable.getSelectedRow(), -1));
 				String dt = '[' + MessageBundle.getMessage("angal.opd.notspecified.msg") + ']';
 				try {
-					DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.ITALIAN);
+					DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.getDefault());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/gui/OpdBrowser.java:384]: Should this be: `new Locale(GeneralData.LANGUAGE)` ?

File: src/main/java/org/isf/opd/gui/OpdEdit.java [hunk 2]
```diff
@@ -143,7 +143,7 @@ private void fireSurgeryUpdated(Opd opd) {
 	private JComboBox diseaseBox = null;
 	private JComboBox diseaseBox2 = null;
 	private JComboBox diseaseBox3 = null;
-	private DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.ITALIAN);
+	private DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.getDefault());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/gui/OpdEdit.java:146]: Should this be: `new Locale(GeneralData.LANGUAGE)` ?

File: src/main/java/org/isf/opd/gui/OpdEditExtended.java [hunk 3]
```diff
@@ -218,7 +218,7 @@ private void fireSurgeryUpdated(Opd opd) {
 	private JLabel jLabelAge = null;
 	private JLabel jLabelSex = null;
 	private GregorianCalendar visitDateOpd = null;
-	private DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.ITALIAN);
+	private DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, Locale.getDefault());
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/opd/gui/OpdEditExtended.java:221]: Should this be: `new Locale(GeneralData.LANGUAGE)` ?

File: src/main/java/org/isf/opd/gui/OpdBrowser.java (hunk 1 above)
Comment (mwithi) [reply src/main/java/org/isf/opd/gui/OpdBrowser.java:384]: I realized that in https://github.com/informatici/openhospital-gui/blob/develop/src/main/java/org/isf/menu/gui/MainMenu.java#L137

we set the default language in the whole application, so from now on we can use .getDefault() method.

we can open an issue to standardize this behavior if necessary, wdyt?

Comment (dbmalkovsky) [reply src/main/java/org/isf/opd/gui/OpdBrowser.java:384]: Good to know (and use in the future); thanks

Comment (mwithi) [reply src/main/java/org/isf/opd/gui/OpdBrowser.java:384]: Thanks to you @dbmalkovsky !

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 4]
```diff
@@ -28,7 +28,7 @@
 		<![CDATA[SELECT TYPE, DATE, IDS, NOTE, DETAILS_1, DETAILS_2, DETAILS_3, DETAILS_4
 FROM (
 	SELECT CONCAT("Opd", " (", OPD_PROG_YEAR, ")") AS TYPE,
-	   OPD_DATE AS DATE, OPD_ID AS IDS,
+	   OPD_DATE_VIS AS DATE, OPD_ID AS IDS,
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:31]:
```suggestion
	   OPD_DATE AS DATE, OPD_ID AS IDS,
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 5]
```diff
@@ -37,15 +37,15 @@ FROM (
 	FROM 
 		OPD 
 		LEFT JOIN PATIENT ON (OPD_PAT_ID = PAT_ID) 
-	WHERE OPD_PAT_ID = $P{patientID} AND (DATE(OPD_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
+	WHERE OPD_PAT_ID = $P{patientID} AND (DATE(OPD_DATE_VIS) BETWEEN $P{Date_from} AND $P{Date_to})
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:40]:
```suggestion
	WHERE OPD_PAT_ID = $P{patientID} AND (DATE(OPD_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 6]
```diff
@@ -129,7 +129,7 @@ UNION
 UNION
 
 	SELECT "Laboratory" AS TYPE,
-	   LAB_DATE AS DATE, LAB_RES AS IDS,
+	   LAB_EXAM_DATE AS DATE, LAB_RES AS IDS,
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:132]:
```suggestion
	   LAB_DATE AS DATE, LAB_RES AS IDS,
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 7]
```diff
@@ -139,13 +139,13 @@ UNION
 		LABORATORY  
 		LEFT JOIN EXAM ON (EXA_ID_A = LAB_EXA_ID_A) 
 		LEFT JOIN LABORATORYROW ON LABR_LAB_ID = LAB_ID
-	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
+	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_EXAM_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:142]:
```suggestion
	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 8]
```diff
@@ -139,13 +139,13 @@ UNION
...
     AND EXA_PROC = 1
     
 UNION
 
 	SELECT "Laboratory" AS TYPE,
-	   LAB_DATE AS DATE, LAB_ID AS IDS,
+	   LAB_EXAM_DATE AS DATE, LAB_ID AS IDS,
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:148]:
```suggestion
	   LAB_DATE AS DATE, LAB_ID AS IDS,
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 9]
```diff
@@ -155,14 +155,14 @@ UNION
 		LABORATORY  
 		LEFT JOIN EXAM ON (EXA_ID_A = LAB_EXA_ID_A) 
 		LEFT JOIN LABORATORYROW ON LABR_LAB_ID = LAB_ID
-	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
+	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_EXAM_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:158]:
```suggestion
	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 10]
```diff
@@ -155,14 +155,14 @@ UNION
...
 	AND EXA_PROC = 2
 GROUP BY LABR_LAB_ID
 
 UNION
 
 	SELECT "Laboratory" AS TYPE,
-	   LAB_DATE AS DATE, LAB_RES AS IDS,
+	   LAB_EXAM_DATE AS DATE, LAB_RES AS IDS,
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:165]:
```suggestion
	   LAB_DATE AS DATE, LAB_RES AS IDS,
```

File: rpt/patient_clinical_sheet_ver3.jrxml [hunk 11]
```diff
@@ -172,7 +172,7 @@ UNION
 		LABORATORY  
 		LEFT JOIN EXAM ON (EXA_ID_A = LAB_EXA_ID_A) 
 		LEFT JOIN LABORATORYROW ON LABR_LAB_ID = LAB_ID
-	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
+	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_EXAM_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```
Comment (mwithi) [inline rpt/patient_clinical_sheet_ver3.jrxml:175]:
```suggestion
	WHERE LAB_PAT_ID = $P{patientID} AND (DATE(LAB_DATE) BETWEEN $P{Date_from} AND $P{Date_to})
```

**Pull request merged**